BaragonData/src/main/java/com/hubspot/baragon/data/BaragonResponseHistoryDatastore.java

```diff
@@ -42,14 +43,16 @@ public void addResponse(String serviceId, String requestId, BaragonResponse resp
     return readFromZk(String.format(RESPONSE_HISTORY_FORMAT, serviceId, requestId), BaragonResponse.class);
   }
 
-  public List<BaragonResponse> getResponsesForService(String serviceId) {
+  public List<BaragonResponse> getResponsesForService(String serviceId, int limit) {
     final List<String> nodes = getChildren(String.format(RESPONSE_HISTORIES_FOR_SERVICE_FORMAT, serviceId));
-    final List<BaragonResponse> responses = Lists.newArrayListWithCapacity(nodes.size());
-    for (String node : nodes) {
+    final List<BaragonResponse> responses = Lists.newArrayListWithCapacity(Math.min(nodes.size(), limit));
+    Iterator<String> nodeIterator = nodes.iterator();
+    for (int i = 0; nodeIterator.hasNext() && i < limit; i++) {
```

Comment:
it'd be preferable to use something like `.subList()` instead of this iterator stuff